setup.py

```diff
@@ -1,5 +1,37 @@
 from setuptools import setup, find_packages
-from dvc import VERSION
+from setuptools.command.build_py import build_py as _build_py
+import os
+
+
+# https://packaging.python.org/guides/single-sourcing-package-version/
+pkg_dir = os.path.dirname(__file__)
+
+# This will define __version__ implicitly
+with open(os.path.join(pkg_dir, "dvc", "version.py")) as fobj:
+    exec(fobj.read())
+
+version = __version__  # noqa: F821
+
+
+# To achieve consistency between the build version and the one provided
+# by your package during runtime, you need to **pin** the build version.
+#
+# This custom class will replace the version.py module with a **static**
+# `__version__` that your package can read at runtime, assuring consistancy.
+#
+# References:
+#   - https://docs.python.org/3.7/distutils/extending.html
+#   - https://github.com/python/mypy
+class build_py(_build_py):
+    def pin_version(self):
+        path = os.path.join(self.build_lib, "dvc")
+        self.mkpath(path)
+        with open(os.path.join(path, "version.py"), "w") as stream:
+            stream.write('__version__ = "{}"\n'.format(version))
```

Comment:
Let's at least put a header comment that this file was autogenerated by `setup.py`. Similar to how we have a comment about it in the current version logic.

Reply:
:+1: